textprocessors.cpp

```diff
@@ -7,7 +7,20 @@
 // Kaleb Text Processors
 
 int count_words(char *text) {
-    return 0;
+    bool startedWord = false;
+    int count = 0;
+    for (int i = 0; text[i] != '\0'; i++) {
+        char currentChar = text[i];
+        if (isspace(currentChar) || ispunct(currentChar)) {
+            if (startedWord) {
+                count++;
+                startedWord = false;
+            }
+        } else if (isalpha(currentChar)) {
+            startedWord = true;
```

Comment:
```suggestion
                startedWord = false;
            }
        } else if (isalpha(currentChar)) {
            count++;
            startedWord = true;
```
If count is done only when there is a space, and the string doesn't end with a space or . the word isn't counted. This method will count the word when we know there's at least a letter.